base/shell/explorer/traywnd.cpp

```diff
@@ -3251,7 +3251,7 @@ class CTrayWindow :
 
     void _Init()
     {
-        m_Position = (DWORD) -1;
```

Comment:
What's wrong with this one ?
the constructor already sets this value to 0, and you're effectively changing behaviour where 0 means ABE_LEFT.

See [this](https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/psdk/shellapi.h;hb=32b0cf6fc6b8f5fab33c10c782f40d36feebbedb#l17) & [that](https://git.reactos.org/?p=reactos.git;a=blob;f=base/shell/explorer/traywnd.cpp;hb=32b0cf6fc6b8f5fab33c10c782f40d36feebbedb#l2078)

Reply:
The constructor initializes it to 0, but CreateTrayWindow then calls _Init, setting it to -1.
I get the following crash.
```
kd> kp
 # Child-SP          RetAddr           Call Site
00 00000000`0012e2f0 00000001`4002ec3b explorer!CTrayWindow::ChangingWinPos(struct _WINDOWPOS * pwp = 0x00000000`0012e8a8)+0x320 [E:\amd64\reactos\base\shell\explorer\traywnd.cpp @ 1381] 
01 00000000`0012e380 00000001`4003029f explorer!CTrayWindow::OnWindowPosChanging(unsigned int uMsg = 0x46, unsigned int64 wParam = 0, int64 lParam = 0n1239208, int * bHandled = 0x00000000`0012e3e4)+0x3b [E:\amd64\reactos\base\shell\explorer\traywnd.cpp @ 2480] 
02 00000000`0012e3b0 00000001`40013da1 explorer!CTrayWindow::ProcessWindowMessage(struct HWND__ * hWnd = 0x00000000`000600c6, unsigned int uMsg = 0x46, unsigned int64 wParam = 0, int64 lParam = 0n1239208, int64 * lResult = 0x00000000`0012e528, unsigned long dwMsgMapID = 0)+0x7ef [E:\amd64\reactos\base\shell\explorer\traywnd.cpp @ 3129] 
03 00000000`0012e480 000007ff`b258eb86 explorer!ATL::CWindowImplBaseT<ATL::CWindow,ATL::CWinTraits<1442840576,0> >::WindowProc(struct HWND__ * hWnd = 0x00000000`000600c6, unsigned int uMsg = 0x46, unsigned int64 wParam = 0, int64 lParam = 0n1239208)+0x121 [E:\amd64\reactos\sdk\lib\atl\atlwin.h @ 1594] 
04 00000000`0012e560 fffff800`00405a50 user32!IntCallWindowProcW(int IsAnsiProc = 0n0, <function> * WndProc = 0x00000000`007e0000, struct _WND * pWnd = 0x00000000`004f9c90, struct HWND__ * hWnd = 0x00000000`000600c6, unsigned int Msg = 0x46, unsigned int64 wParam = 0, int64 lParam = 0n1239208)+0x7e6 [E:\amd64\reactos\win32ss\user\user32\windows\message.c @ 1547] 
05 fffff880`044e4690 fffff880`044e4690 nt!KiCallUserMode
```

Reply:
I added a check for -1 instead. This also fixes the issue.

Reply:
Nice.
The -1 is more or less a placeholder for 'Should not use this yet'.
Due to the message passing order this now breaks, ideally all access should use one common accessor that checks for `-1`.
This sounds like a nice starter-task for jira or so.